gateway-util-launcher/src/test/java/org/apache/knox/gateway/launcher/ExtenderTest.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.knox.gateway.launcher;
+
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Properties;
+
+import static org.junit.Assert.assertEquals;
+
+public class ExtenderTest {
+
+    private Path tempDir;
+    private Path confDir;
+    private Path configFilePath;
+
+    @Test
+    public void extendClassPathPropertyTest() throws IOException {
+        this.setupDirs();
+        Properties properties = new Properties();
+        properties.setProperty("class.path", "classpath");
+        properties.setProperty("main.class", "org.apache.knox.gateway.GatewayServer");
+        Extender extender = new Extender(confDir.toFile(), properties);
+
+        String configContent = this.getConfigContent("/new/classp/*");
+        Files.write(configFilePath, configContent.getBytes(StandardCharsets.UTF_8));
+        extender.extendClassPathProperty();
+
+        assertEquals("/new/classp/*;classpath", properties.getProperty("class.path"));
+        this.cleanUpDirs();
```

Comment:
What if an exception occurred before this line? No cleanup?
Why not annotate `setupDirs` and `cleanUpDirs` with `Before/After`? Even if they are not needed for some of the test cases, they don't cause any harm, and error handling is managed by OOTB.
What do you think?

Reply:
Yes, that was my thinking. They are not needed for every case, however I didn't think of error handling. I changed them to `After/Before`.